packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts

```diff
@@ -1442,6 +1442,8 @@ export class KeyboardNavigationController extends modules.ViewController {
             const isFocusedElementDefined = isElementDefined(
               $focusedElementInsideCell,
             );
+            const columnIndex = this._rowsView.getCellIndex($cell);
+            const column = this.getController('columns').getVisibleColumns()[columnIndex];
```

Comment:
I found it is needed to take into account the [columnIndexOffset](https://github.com/DevExpress/DevExtreme/blob/1db274a1b82f1eafb565c0dd82bb0318b4aa1c87/packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts#L854) when you take a column.

Reply:
columnIndexOffset is used depending on whether we get all the visible columns or only the rendered ones.
Here's a different case getting only the rendered ones: [m_keyboard_navigation.ts#L1211](https://github.com/DevExpress/DevExtreme/blob/1db274a1b82f1eafb565c0dd82bb0318b4aa1c87/packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts#L1211) - here columnIndexOffset is not used

Anyway, I refactored what I can, except this part: https://github.com/DevExpress/DevExtreme/pull/26149/files#diff-cb3dbff86c1db4ac00da48f26a0f18d218f4e696d3b3b1e75785d419be1bf327R1831
since 'columnIndex' is used later in the code